api/src/libs/migrationUtils.ts

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { BusinessesDataClient, UserDataClient } from "@domain/types";
+import { Business, CURRENT_VERSION, UserData } from "@shared/userData";
+
+export async function processBusinessesMigration(
+  userData: UserData,
+  businessesData: Record<string, Business>,
+  userDataClient: UserDataClient,
+  businessesDataClient: BusinessesDataClient,
+  logger: any
+): Promise<void> {
+  for (const businessId in businessesData) {
+    logger.LogInfo(`Starting migration for business ID ${businessId} for user ${userData.user.id}`);
+    const businessData = businessesData[businessId];
+    try {
+      const existingBusiness = await businessesDataClient.get(businessId);
+      logger.LogInfo(
+        existingBusiness
+          ? `Found existing business data for business ID ${businessId}, current version: ${existingBusiness.version}`
+          : `No existing business found for business ID ${businessId}, setting to default version`
+      );
+      const currentVersion = existingBusiness ? existingBusiness.version + 1 : CURRENT_VERSION;
```

Comment:
This should check if existingBusiness.version exists, so `existingBusiness.version ? ....`